src/coreclr/vm/interpexec.cpp

```diff
@@ -1835,9 +1890,25 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
 
                 case INTOP_CALL_HELPER_P_P:
                 {
-                    HELPER_FTN_P_P helperFtn = GetPossiblyIndirectHelper<HELPER_FTN_P_P>(pMethod, ip[2]);
                     void* helperArg = pMethod->pDataItems[ip[3]];
 
+                    MethodDesc *pILTargetMethod = NULL;
+                    HELPER_FTN_P_P helperFtn = GetPossiblyIndirectHelper<HELPER_FTN_P_P>(pMethod, ip[2], &pILTargetMethod);
+                    if (pILTargetMethod != NULL)
+                    {
+                        returnOffset = ip[1];
+                        int stackOffset = pMethod->allocaSize;
+                        callArgsOffset = stackOffset;
+
+                        // Pass argument to the target method
+                        LOCAL_VAR(stackOffset, void*) = helperArg;
+
+                        targetMethod = pILTargetMethod;
+                        ip += 4;
+                        goto CALL_INTERP_METHOD;
+                    }
```

Comment:
It would be nice if we could encapsulate this repeated logic somehow since it's fairly nuanced but I don't see an easy way to do it since we're populating locals and doing a goto...

Reply:
We discussed that with JanV before and he didn't like the macros either, because they are difficult to debug.